writer.go

```diff
@@ -284,6 +297,66 @@ func (w *GenericWriter[T]) File() FileView {
 	return w.base.File()
 }
 
+// ConcurrentRowGroupWriter is a row group writer that can be used to write row groups
+// in parallel. Multiple row groups can be created concurrently and written to independently,
+// but they must be committed serially to maintain the order of row groups in the file.
+//
+// See BeginRowGroup for more information on how this can be used.
+//
+// While multiple row groups can be created concurrently, a single row group must be written
+// sequentially.
+type ConcurrentRowGroupWriter interface {
+	RowWriterWithSchema
+
+	// Flush flushes any buffered data in the row group's column writers.
+	// This could be called before Commit to ensure all data pages are flushed.
+	Flush() error
+
+	// ColumnWriters returns the column writers for this row group, allowing
+	// direct access to write values to individual columns.
+	ColumnWriters() []*ColumnWriter
```

Comment:
What do we need to expose this method for? I don't see any tests showing how it would be employed.